### PR TITLE
Allow changing template type by admin

### LINKED ISSUE #835: [FEATURE] server: update template to another template type
**Is your feature request related to a problem? Please describe.**
A new feature implemented in https://github.com/apache/cloudstack/pull/3945 needs primate implementation

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by ravening**
Fixes #835 

Currently, the template has the following types:
'BUILTIN', 'USER', 'SYSTEM', 'ROUTING'

Provide ui support for admins os that they can change
the template type

Screenshot 2020-10-27 at 19 10 09

Template type can be seen in template details also

Screenshot 2020-10-27 at 19 10 28

### CODE REVIEW

**Comment by rhtyd** [comment]
Does this depend on any backend API change?

**Comment by rhtyd** [comment]
Nevermind - I checked, it does; we may need to wait until the upstream PR is merged at all.

File: src/config/section/image.js
Comment (rhtyd) [inline src/config/section/image.js:51]: @ravening cc @weizhouapache @DaanHoogland can you check if this causes regression wrt legacy UI? The is public/routing/features and `url` were previously accepted by old UI as well. What about upload template API/form?

File: src/config/section/image.js
Comment (rhtyd) [inline src/config/section/image.js:105]: this LGTM

File: src/config/section/image.js
Comment (rhtyd) [inline src/config/section/image.js:114]: What is PERHOST @ravening @weizhouapache ?

File: src/config/section/image.js
Comment (ravening) [reply src/config/section/image.js]: `ispublic` and `isfeatured` is displayed for regular users also. so made the change

File: src/config/section/image.js
Comment (ravening) [reply src/config/section/image.js]: @rhtyd removed `PERHOST` as its not needed

**Comment by GabrielBrascher** [comment]
@blueorangutan package

**Comment by GabrielBrascher** [comment]
It fails to open the edit popup when trying to edit a template. Tested via http://primate-qa.cloudstack.cloud:8080/client/pr/838.

Can you please check this @ravening?

**Comment by ravening** [comment]
> It fails to open the edit popup when trying to edit a template. Tested via http://primate-qa.cloudstack.cloud:8080/client/pr/838.
> 
> 
> 
> Can you please check this @ravening?

@GabrielBrascher do you have the back end change as well?
I tested with having backend change and worked fine. I have attached the screenshot as well

**Comment by GabrielBrascher** [comment]
Ah ... it got merged yesterday (https://github.com/apache/cloudstack/pull/3945), I see. I will test it later then based on the recente merged commit.

**Comment by ravening** [comment]
> > It fails to open the edit popup when trying to edit a template. Tested via http://primate-qa.cloudstack.cloud:8080/client/pr/838.
> > Can you please check this @ravening?
> 
> @GabrielBrascher do you have the back end change as well?
> I tested with having backend change and worked fine. I have attached the screenshot as well

@GabrielBrascher on a side note, Im curious to know how to have multiple instances on primate running for each pr? is it an automated task? is there a way I can do that on my setup as well? I see that for each pr, there is an endpoint /client/pr/<pr#> . I want to know how to generate new setup like that for each pr

**Comment by GabrielBrascher** [comment]
@ravening I have not much idea on the _blueoragutan_ configuration. @rhtyd or @PaulAngus would be able to give some details.

My guess is that this was configured via Jenkins in order to provide the same _Mocked_ infra (hosts, zone, networks, VMs, etc) for all PRs that are tested.

**Review by GabrielBrascher — APPROVED** [review]
LGTM based on code and test.

**Comment by PaulAngus** [comment]
Blue Orangutan sends build requests to ShapeBlue's internal infra to do builds;  So calls to it need to come from ShapeBlue people to protect the infra (and it's easier to shout at them than community members :) ).

We can happily share the actual build process (done via Jenkins).

The medium/long term goal is to have distributed build and test environments across 'donated' central community hardware and private environments, which would be orchestrated from a central location, with some kind of deployment planner.

Internally we know this project as Winston ("I fix things"), and we're going to create it internally which we will then share as a draft blueprint to be taken forward as a community (assuming people want it),

**Comment by ravening** [comment]
> Blue Orangutan sends build requests to ShapeBlue's internal infra to do builds; So calls to it need to come from ShapeBlue people to protect the infra (and it's easier to shout at them than community members :) ).
> 
> We can happily share the actual build process (done via Jenkins).
> 
> The medium/long term goal is to have distributed build and test environments across 'donated' central community hardware and private environments, which would be orchestrated from a central location, with some kind of deployment planner.
> 
> Internally we know this project as Winston ("I fix things"), and we're going to create it internally which we will then share as a draft blueprint to be taken forward as a community (assuming people want it),

@PaulAngus Can you please share the build process and steps so that I will try implementing the same for our internal infrastructure?

**Comment by PaulAngus** [comment]
@ravening 

I've checked with Rohit regarding the base VM that we use to build primate.

It can be built using any Ubuntu LTS 18.04/20.04 base docker container. 

There are  build instructions are in the repo's README (_https://github.com/apache/cloudstack-primate#getting-started_)
These are wrapped up in `https://github.com/apache/cloudstack-primate/blob/master/packaging/package.sh

The overview of installation (again thanks to Rohit) is:

Install nodejs LTS

```
curl -sL https://deb.nodesource.com/setup_12.x | sudo -E bash -
sudo apt-get install -y nodejs debhelper rpm
```
Install build tools

`npm install -g @vue/cli webpack eslint`

In a cloned Primate repo and run package.sh

```
cd packaging
bash -x package.sh
```
In the README.md - _https://github.com/apache/cloudstack-primate#packaging_

**Comment by rhtyd** [comment]
Looks like related backend PR was merged, should we consider merging this @PaulAngus @DaanHoogland @davidjumani ?

@blueorangutan package

**Review by davidjumani — APPROVED** [review]
I'm for merging this

**Pull request merged**
